File: docs-old/notintoc/codesnippet/CSharp/how-to--program-the-progress-bar-region-of-the-status-bar_1.cs
        void ProgressBarExample()
        {
            IVsStatusbar statusBar =
                (IVsStatusbar)GetService(typeof(SVsStatusbar));
            uint cookie = 0;
            string label = "Progress bar label...";

            // Initialize the progress bar.
            statusBar.Progress(ref cookie, 1, "", 0, 0);

            for (uint i = 0, total = 100; i <= total; i++)
            {
                // Display incremental progress.
                statusBar.Progress(ref cookie, 1, label, i, total);
                System.Threading.Thread.Sleep(1);
            }

            // Clear the progress bar.
            statusBar.Progress(ref cookie, 0, "", 0, 0);
        }

File: docs-old-1102/misc/codesnippet/VisualBasic/how-to-program-the-progress-bar-region-of-the-status-bar_1.vb
    Private Sub ProgressBarExample()
        Dim statusBar As IVsStatusbar = DirectCast(GetService(GetType(SVsStatusbar)), IVsStatusbar)
        Dim cookie As UInteger = 0
        Dim label As String = "Progress bar label..."

        ' Initialize the progress bar. 
        statusBar.Progress(cookie, 1, "", 0, 0)

        Dim i As UInteger = 0, total As UInteger = 100
        While i <= total
            ' Display incremental progress. 
            statusBar.Progress(cookie, 1, label, i, total)
            System.Threading.Thread.Sleep(1)
            i += 1
        End While

        ' Clear the progress bar. 
        statusBar.Progress(cookie, 0, "", 0, 0)
    End Sub